core/src/test/java/org/infinispan/security/CacheAuthorizationTest.java

```diff
@@ -91,6 +96,35 @@ public Void run() {
       });
    }
 
+   public void testAdminCombinations() throws Exception {
+      List<Runnable> calls = Arrays.asList(
+            () -> cacheManager.getGlobalComponentRegistry(),
+            () -> cacheManager.getCacheManagerConfiguration());
+
+      for (final AuthorizationPermission perm : AuthorizationPermission.values()) {
+         for (Runnable fn : calls) {
+
+            PrivilegedExceptionAction<Boolean> action = () -> {
+              fn.run();
+              return true;
+            };
+
+            // only admin must work
+            if (AuthorizationPermission.ADMIN.equals(perm)) {
+               assertTrue(Security.doAs(SUBJECTS.get(perm), action));
+            } else {
+               // other roles must fail
+               try {
+                  Security.doAs(SUBJECTS.get(perm), action);
```

Comment:
You're not really asserting that the exception is thrown, you're only checking the exception type **if** the exception is thrown. You should use 

```
               Exceptions.expectException(PrivilegedActionException.class, SecurityException.class,
                                          () -> Security.doAs(subject, action));
```

which fails for permission `ALL`, because `ALL` includes `ADMIN` (unless you apply the suggestion on line 113)